src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,180 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node head;
+    private Node tail;
+
     @Override
     public void add(T value) {
+        if (size < 2) {
+            addHead(value);
+            return;
+        }
+
+        Node node = new Node<>(tail, value, null);
+        tail.next = node;
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0) {
+            throw new IndexOutOfBoundsException("index " + index + " is out of size " + size);
+        }
+
+        if (size == 0 || index == size) {
+            add(value);
+            return;
+        }
+
+        if (index == 0) {
+            addIfIndex0(value, index);
+            return;
+        }
+
+        checkIndex(index);
+        Node indexNode = getNode(index);
+        Node insertNode = new Node(indexNode.prev, value, indexNode);
+        indexNode.prev.next = insertNode;
+        indexNode.prev = insertNode;
+
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).value;
```

Comment:
This is code repeating. You always check the index before getting node by index. What you should do is move the check logic INSIDE the get method.